fedimint-client/src/lib.rs

```diff
@@ -1,3262 +1,11 @@
-#![deny(clippy::pedantic)]
```

Comment:
Rename, with a tiny stub to forward everything back to `fedimint-client-module`.